default/python/AI/InvasionAI.py

```diff
@@ -76,15 +78,25 @@ def get_invasion_fleets():
             planet_partial_vis_turn = universe.getVisibilityTurnsMap(pid, empire_id).get(fo.visibility.partial, -9999)
             if planet_partial_vis_turn < sys_partial_vis_turn:
                 continue
+            best_base_planet = INVALID_ID
+            best_trooper_count = 1  # won't bother using base troopers unless better than 1 per ship
```

Comment:
Not sure I like this.

Assume we have only 1 trooper per part. Still, if we want to invade a planet with, say, 6 troops, there is not really an argument to build a bigger ship (6 slots) somewhere else which might need to be moved here.

IMO, the ```if n_bases > 20:``` check should be sufficient to rule out inefficient cases.